src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -38,12 +47,59 @@ public LogicManager(Model model, Storage storage) {
         gradPadParser = new GradPadParser();
     }
 
+    /**
+     * Handles a stalled situation where a command requires confirmation from the user to execute.
+     *
+     * @param command The command to be stalled.
+     * @param commandText The user input text to be stalled.
+     * @return A CommandResult requesting confirmation from the user.
+     * @throws CommandException if a requested DeleteCommand is invalid.
+     */
+    public CommandResult handleStall(Command command, String commandText) throws CommandException {
+        if (command instanceof ClearCommand) {
+            assignStalledComponents(command, commandText);
+            return new CommandResult(ClearCommand.MESSAGE_CONFIRMATION + MESSAGE_CONFIRMATION_SYNTAX);
+        } else {
+            Module moduleToBeDeleted = ((DeleteCommand) command).getModuleToDelete(model);
+            assignStalledComponents(command, commandText);
+            return new CommandResult(DeleteCommand.MESSAGE_CONFIRMATION + moduleToBeDeleted
+                    + MESSAGE_CONFIRMATION_SYNTAX);
+        }
+    }
+
+    /**
+     * Assigns the specified command and commandText to the relevant stalled components.
+     *
+     * @param command The command to be stalled.
+     * @param commandText The user input text to be stalled.
+     */
+    public void assignStalledComponents(Command command, String commandText) {
```

Comment:
Really really small nit: I think this method can remain as private since only `handleStall` should use it HAHA Actually how come it's changed from private -> public since the last review ah?

Reply:
HAHAHAHA i thought like with javadocs would be nicer LOL i'll remove!